src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts

```diff
@@ -117,40 +130,79 @@ export const useTradeExecution = ({
         tradeQuote,
         chainId,
         accountMetadata,
-        stepIndex,
+        stepIndex: activeStepOrDefault,
         supportsEIP1559,
         buyAssetUsdRate,
         feeAssetUsdRate,
       },
     )
 
-    const sellTxHash = await swapper.executeTrade({
+    sellTxHashRef.current = await swapper.executeTrade({
       txToSign: unsignedTxResult,
       wallet,
       chainId,
     })
 
-    setSellTxHash(sellTxHash)
+    setSellTxHash(sellTxHashRef.current)
 
     await poll({
       fn: async () => {
+        // This should never happen, but TS mang
+        if (!sellTxHashRef.current) return
         const { status, message, buyTxHash } = await swapper.checkTradeStatus({
           quoteId: tradeQuote.id,
-          txHash: sellTxHash,
+          txHash: sellTxHashRef.current,
           chainId,
         })
 
-        setMessage(message)
-        setBuyTxHash(buyTxHash)
-        setStatus(status)
+        // TODO(gomes): do we want to bring in the concept of watching for a step execution in addition to trade execution?
+        // useTradeExecution seems to revolve around the idea of a holistic trade execution i.e a sell/buy asset for the whole trade,
+        // but we may want to make this granular to the step level?
+        if (isLastStepRef.current || status === TxStatus.Failed) {
+          setMessage(message)
+          setBuyTxHash(buyTxHash)
+          setTradeStatus(status)
+        }
+
+        // Tx confirmed/pending for a mid-trade hop, meaning the trade is still pending holistically
+        else if (status === TxStatus.Confirmed || status === TxStatus.Pending) {
+          setTradeStatus(TxStatus.Pending)
+        }
 
         return status
       },
-      validate: status => status === TxStatus.Confirmed || status === TxStatus.Failed,
+      validate: status => {
+        return status === TxStatus.Confirmed || status === TxStatus.Failed
```

Comment:
Updated to debug, but might as well in case we need to debug this again, which we'll probably do